opengovsg/plumber
# PLU-81: [IF-THEN-19] ***HALP*** HACKFIX default rows not showing in multi-row

## Problem
We always add a single default empty row to multi-row fields when adding new actions that have such fields. But for some reason, multi-row fields are not showing this default field unless 

To reproduce this - just create an If-then step, and you'll see that the "Conditions" field is empty when it should have a single empty row. But if you expand "Test Action" and then re-expand "Set up action"... the field shows up!

## Solution
I have banged my head against this for an entire day, but I can't find a way to fix it properly. The root cause appears to be `useFieldArray` lagging behind the actual field data by a single render, but only if the default value was used (_if an if-then step already has conditions set, this problem does not occur_).

The best I can do is a hackfix; we will render the underlying field data directly if the rows from `useFieldArray` is empty. This way, we can't do any worse (I'll be very surprised if useFieldArray caches its rows and doesn't write it to the underlying field data).

**BEFORE**:
<img width=500 src="https://github.com/opengovsg/plumber/assets/135598754/7018e207-1dfd-4c1b-a95c-88cbc58fa055" />

**AFTER**:
<img width=500 src="https://github.com/opengovsg/plumber/assets/135598754/513e8394-52be-4efd-a95d-ae5b5458c48b" />


## Tests
- Check that default rows now render on 1st render, without requiring re-mount via collapsing and expanding.
- Check that values are saved correctly in the DB.
- Check that saved values are restored correctly after a refresh.
